__main__.py

```diff
@@ -86,38 +93,29 @@ def conversation_name(conv):
     return ', '.join(u.full_name for u in users)
 
 
-def receive_message_callback(message_event, client, conversation):
-    def receive_message():
-        """
-        For a chat message event, send a response reflecting the same message.
-        """
-
-        segments = hangups.ChatMessageSegment.from_str(
-            '[FROM HANGBOT] {}'.format(message_event.text))
-        conversation.send_message(segments)
-        (asyncio.async(conversation.send_message(segments))
-            .add_done_callback(on_message_sent))
-
-    def on_message_sent(future):
-        """Handle showing an error if a message fails to send."""
-
-        try:
-            future.result()
-        except hangups.NetworkError as e:
-            logger.exception('Failed to send message: %s', e)
+async def receive_message(message_event, client, conversation):
+    """
+    For a chat message event, go through the enabled scripts until one of
+    them processes a message.
+    """
 
-    return receive_message
+    try:
+        for script in ENABLED_SCRIPTS:
+            if await script.process(client, conversation, message_event):
```

Comment:
I think it could be a little clearer to name it `process_message` and then order the arguments in the same way as `receive_message`.